lib/txinput.js

```diff
@@ -149,11 +166,38 @@ class LedgerTXInput {
 
   getPrev() {
     if (!this._prev)
-      this._prev = this.tx.outputs[this.index].script;
+      this._prev = this.getCoin().script;
 
     return this._prev;
   }
 
+  /**
+   * Get pubkey script or redeem script
+   * also resolve Witness
+   * @returns {Boolean}
+   */
+
+  getPrevRedeem() {
+    if (this._prevred)
+      return this._prevred;
+
+    const prev = this.getPrev();
+
+    if (prev.isScripthash() && this.witness && !this.redeem) { // nested
+      const wpkh = this.getRing().getProgram().getWitnessPubkeyhash();
+      this._prevred = Script.fromPubkeyhash(wpkh);
+    } else if (this.isScripthash()) { // witness or not, we need redeem script
+      this._prevred = this.redeem;
```

Comment:
Ah, `prevred` === redeem script for prev outpoint. How do we label this in the normal wallet?

Reply:
This is different, this is either previous output scirpt or redeem script.